core/validator_manager.go

```diff
@@ -0,0 +1,151 @@
+package core
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"sync"
+
+	"github.com/0xPolygon/go-ibft/messages/proto"
+)
+
+var (
+	errVotingPowerNotCorrect = errors.New("total voting power is zero or less")
+)
+
+// ValidatorBackend defines interface that has GetVotingPower
+type ValidatorBackend interface {
+	// GetVotingPowers returns map of validators addresses and their voting powers for the specified height.
+	GetVotingPowers(height uint64) (map[string]*big.Int, error)
+}
+
+// ValidatorManager keeps voting power and other informations about validators
+type ValidatorManager struct {
+	vpLock *sync.RWMutex
+
+	// quorumSize represents quorum for the height specified in the current View
+	quorumSize *big.Int
+
+	// validatorsVotingPower is a map of the validator addresses on their voting power for
+	// the height specified in the current View
+	validatorsVotingPower map[string]*big.Int
+
+	backend ValidatorBackend
+
+	log Logger
+}
+
+// NewValidatorManager creates new ValidatorManager
+func NewValidatorManager(backend ValidatorBackend, log Logger) *ValidatorManager {
+	return &ValidatorManager{
+		quorumSize:            big.NewInt(0),
+		backend:               backend,
+		validatorsVotingPower: nil,
+		log:                   log,
+		vpLock:                &sync.RWMutex{},
+	}
+}
+
+// Init sets voting power and quorum size
+func (vm *ValidatorManager) Init(height uint64) error {
+	vm.vpLock.Lock()
+	defer vm.vpLock.Unlock()
+
+	validatorsVotingPower, err := vm.backend.GetVotingPowers(height)
+	if err != nil {
+		return err
+	}
+
+	totalVotingPower := calculateTotalVotingPower(validatorsVotingPower)
+	if totalVotingPower.Cmp(big.NewInt(0)) <= 0 {
+		return errVotingPowerNotCorrect
+	}
+
+	vm.validatorsVotingPower = validatorsVotingPower
+	vm.quorumSize = calculateQuorum(totalVotingPower)
+
+	return nil
+}
+
+// HasQuorum provides information on whether messages have reached the quorum
+func (vm *ValidatorManager) HasQuorum(addressMap map[string]struct{}) bool {
```

Comment:
Rename `addressMap` to `sendersAddrs`. WDYT?

Reply:
fixed https://github.com/0xPolygon/go-ibft/commit/f73123680449e1b37230f7bd351a80fc63464f98